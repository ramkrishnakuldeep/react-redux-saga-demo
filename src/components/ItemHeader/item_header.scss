.item_header {
    position: fixed;
    height: 60px;
    width: 100%;
    top: 0;
    background: dodgerblue;
    display: flex;
    align-items: center;

    &>span {
        flex-grow: 2;
        color: lightsteelblue;
    }

    svg {
        width: 40px;
        padding: 20px;
        color: lightsteelblue;
    }

    .name {
        font-size: 2rem;
        text-transform: capitalize;
    }
}