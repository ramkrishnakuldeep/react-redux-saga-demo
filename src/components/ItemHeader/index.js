import React from 'react';
import PropTypes from 'prop-types';
import { item } from '../../propTypes';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronLeft, faSearch, faShoppingCart } from '@fortawesome/free-solid-svg-icons';
import './item_header.scss';

const ItemHeader = ({ item }) => {
    return (
        <header className="item_header">
            <Link to={'/'}>
                <FontAwesomeIcon icon={faChevronLeft} />
            </Link>
            <span className="name"> {item.name} </span>
            <FontAwesomeIcon icon={faSearch} />
            <FontAwesomeIcon icon={faShoppingCart} />
        </header>
    );
};

ItemHeader.defaultProps = {
    item: {}
};

ItemHeader.propTypes = {
    item: PropTypes.shape(item)
};

export default ItemHeader;
