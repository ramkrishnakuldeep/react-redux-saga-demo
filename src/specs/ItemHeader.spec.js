import React from 'react';
import { shallow } from '../enzyme';
import ItemHeader from '../components/ItemHeader';

describe('ItemHeader component', () => {
    it('initial render', () => {
        const component = shallow(<ItemHeader />);
        expect(component).toMatchSnapshot();
    });
});
