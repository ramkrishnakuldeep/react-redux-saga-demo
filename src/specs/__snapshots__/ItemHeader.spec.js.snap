// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemHeader component initial render 1`] = `
<header
  className="item_header"
>
  <Link
    to="/"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            320,
            512,
            Array [],
            "f053",
            "M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z",
          ],
          "iconName": "chevron-left",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </Link>
  <span
    className="name"
  >
     
     
  </span>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f002",
          "M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z",
        ],
        "iconName": "search",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          576,
          512,
          Array [],
          "f07a",
          "M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z",
        ],
        "iconName": "shopping-cart",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</header>
`;
